### PR TITLE
docs(doxygen): Add doxygen conf file and documentations

### PR DESCRIPTION
**Pull request by GMishx**
## Changes
1.  Created Doxygen configuration file (`fossolog_doxygen.conf`) required to create doxygen documents.
1.  Changes comments on source pages as per the doxygen format.

Current documentation: https://gmishx.github.io/ (updated on every commit).

### CODE REVIEW

File: fossology_doxygen.conf [hunk 1]
```diff
@@ -0,0 +1,2427 @@
+# Doxyfile 1.8.11
```
Comment (maxhbr) [inline fossology_doxygen.conf]: Missing licese header

Comment (GMishx) [reply fossology_doxygen.conf]: Added the header. Please check.

File: src/maintagent/agent/maintagent.h [hunk 2]
```diff
@@ -33,6 +33,9 @@
 #include <libfossology.h>
 #define FUNCTION
 
+/**
+ * Maximum buffer to use
```
Comment (mcjaeger) [inline src/maintagent/agent/maintagent.h]: actually, ok, but not so much informative. I understand that there is not more information currently, but ...

File: src/mimetype/agent/finder.c [hunk 3]
```diff
@@ -244,14 +241,17 @@ int DBCheckFileExtention()
...
 
 /**
- * \brief get the ID for the default mimetype.
+ * \brief Get the ID for the default mimetype.
+ *
  *  Options are:
- *  application/x-empty      :: zero-length file
- *  text/plain               :: 1st 100 characters are printable
- *  application/octet-stream :: 1st 100 characters contain binary
+ *  | Mimetype                 | Description |
```
Comment (mcjaeger) [inline src/mimetype/agent/finder.c]: fancy!

File: src/scheduler/agent/logging.h [hunk 4]
```diff
@@ -54,13 +61,13 @@ extern log_t* main_log;
 
 /*
  * The following macro definitions are meant to act as their own statement in
```
Comment (mcjaeger) [inline src/scheduler/agent/logging.h]: nice catches

File: src/ununpack/agent_tests/Unit/test_FindCmd.c [hunk 5]
```diff
@@ -247,7 +308,11 @@ void testFindCmd4NtfsFile()
 }
 
 /**
- * @brief find partition
+ * \brief find partition file
+ * \test
+ * -# Call FindCmd() on parition file
```
Comment (mcjaeger) [inline src/ununpack/agent_tests/Unit/test_FindCmd.c]: finally! found a spelling mistake!

File: src/ununpack/agent_tests/Unit/test_TraverseStart.c [hunk 6]
```diff
@@ -55,7 +64,10 @@ void testTraverseStartNormal()
 }
 
 /**
- * @brief test traversestart dirctory
+ * @brief test TraverseStart() dirctory
```
Comment (mcjaeger) [inline src/ununpack/agent_tests/Unit/test_TraverseStart.c]: uuhhh, second one!

**Comment by mcjaeger** [comment]
@maxhbr sorry for overriding, hope it is OK, since lic header is added.

**Comment by mcjaeger** [comment]
tested-by:michael.c.jaeger@siemens.com - was testing if the fossology works in vagrant actually, and looked at doxy web pages.

File: src/ununpack/agent_tests/Unit/test_FindCmd.c (hunk 5 above)
Comment (GMishx) [reply src/ununpack/agent_tests/Unit/test_FindCmd.c]: Fixed in #1174

File: src/ununpack/agent_tests/Unit/test_TraverseStart.c (hunk 6 above)
Comment (GMishx) [reply src/ununpack/agent_tests/Unit/test_TraverseStart.c]: Fixed in #1174

**Comment by GMishx** [comment]
Pushed the HTMLs to https://fossology.github.io/

**Pull request merged**